project-repo/code-practices.md

```diff
@@ -0,0 +1,41 @@
+# Code practices
+
+- All new code comes with tests. Fixes not coming with tests is rather the exception than the norm.
+
+- We follow upstream/community formatting conventions for the given technology. For instance. in Go, we tighten it using `gofumt` instead of `gofmt`. In Rust, `rustfmt`.
+
+- The code should always be linting-error free.
+
+- We should not have any warning/error when building a project. Any of those needs to be "clear" (with a comment explaining why we are ignoring this warning).
+
+- Every method have a docstring/annotation explaining what the function does, what it accepts and what it returns. Also, for languages raising exceptions, if any kind of specific exception that this method raises.
```

Comment:
The second sentence is not complete? :) (could be a copy of my google doc and I was interrupted in the middle of the sentence). I think the idea is to document what exceptions types the method can raise for those languages.

Reply:
Btw: for simple private method/function, the docstring requirement (if obvious) can be leverage in some cases IMHO. Like an add method which doesn’t need return any method doesn’t really need a doc comment.